ui/src/components/MessageSnackBar.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+  <!-- display a message or error to the end user, and optionally perform a customized action
+    usage: <MessageSnackBar v-bind:bus="bus"></MessageSnackBar>
+      where bus is a vue instance.
+    To display the snackbar, do:
+      bus.$emit('message', config: {
+        content: '',
+        noTranslate: false,
+        action: {
+          title: '',
+          func: (vm) => {...}
+        })
+    can replace 'message' with 'error' or 'notification' for different style of messages.
+    in config, 
+      'content' is the content to be displayed, the component will try to localize the string based on current locale if possible.
+      setting a truthy value to 'noTranslate' will disable translation
+      setting 'action' to false will disable the action button (only show 'dismiss' in the snackbar)
+      'action.title' is the text to be displayed on the button
+      'action.func' is a function to perform the desired action when the button is clicked, the current Vue instance
+        is passed to the function to access relevant context
+    for reference of the event bus model, see https://code.luasoftware.com/tutorials/vuejs/parent-call-child-component-method/
+    -->
+  <div>
+    <!-- TODO: add multiple messages support? 
```

Comment:
Have you had more thoughts about this? I'd say leave the `TODO` in the code and we can come back to it later if this becomes an issue.

Reply:
I think programmatically it is possible to keep a reference for each of the snackbar displayed, and remove them when they disappear. But the `SnackBar` component itself can't re-align when there is already a snackbar behind it. So it will probably look similar to how it does now, with one snackbar on top of multiple ones.